apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.Data/SpannerCommand.cs

```diff
@@ -0,0 +1,367 @@
+            var mutations = new List<Mutation>();
+            if (SpannerCommandTextBuilder.SpannerCommandType != SpannerCommandType.Delete)
+            {
+                var w = new Mutation.Types.Write {Table = SpannerCommandTextBuilder.TargetTable};
+                w.Columns.AddRange(Parameters.Cast<SpannerParameter>().Select(x => x.SourceColumn ?? x.ParameterName));
+                w.Values.Add(new ListValue {
+                    Values = {Parameters.Cast<SpannerParameter>().Select(x => TypeMap.ToValue(x.Value, x.TypeCode))}
+                });
+                switch (SpannerCommandTextBuilder.SpannerCommandType)
+                {
+                    case SpannerCommandType.Update:
+                        mutations.Add(new Mutation {
+                            Update = w
+                        });
+                        break;
+                    case SpannerCommandType.Insert:
+                        mutations.Add(new Mutation {
+                            Insert = w
+                        });
+                        break;
+                    case SpannerCommandType.InsertOrUpdate:
+                        mutations.Add(new Mutation {
+                            InsertOrUpdate = w
+                        });
+                        break;
+                    default:
+                        throw new ArgumentOutOfRangeException();
+                }
+            }
+            else
+            {
+                var w = new Mutation.Types.Delete {Table = SpannerCommandTextBuilder.TargetTable};
+                w.KeySet.Keys.Add(new ListValue {
+                    Values = {Parameters.Cast<SpannerParameter>().Select(x => TypeMap.ToValue(x.Value, x.TypeCode))}
+                });
+                switch (SpannerCommandTextBuilder.SpannerCommandType)
+                {
+                    case SpannerCommandType.Delete:
+                        mutations.Add(new Mutation {
+                            Delete = w
+                        });
+                        break;
+                    default:
+                        throw new ArgumentOutOfRangeException();
+                }
+            }
+
+            // Make the request.  This will commit immediately or not depending on whether a transaction was explicitly created.
+            await GetSpannerTransaction().ExecuteMutationsAsync(mutations, cancellationToken)
+                .WithTimeout(TimeSpan.FromSeconds(CommandTimeout), "The timeout of the SpannerCommand was exceeded.");
+            // Return the number of records affected.
+            return mutations.Count;
+        }
+
+        /// <inheritdoc />
+        public override object ExecuteScalar()
+        {
+            return ExecuteScalarAsync(_synchronousCancellationTokenSource.Token).Result;
+        }
+
+        /// <summary>
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <returns></returns>
+        public Task<T> ExecuteScalarAsync<T>()
```

Comment:
"None" is not a constant and while I could use default(CancellationToken), thats assuming the value of None, which from a purist perspective isnt safe.
Also, the base class has separate overloads instead of optional parameters, so being consistent isnt terrible.